presto-spi/src/main/java/com/facebook/presto/spi/plan/JoinNode.java

```diff
@@ -118,20 +108,19 @@ public JoinNode(
         this.type = type;
         this.left = left;
         this.right = right;
-        this.criteria = ImmutableList.copyOf(criteria);
-        this.outputVariables = ImmutableList.copyOf(outputVariables);
+        this.criteria = unmodifiableList(new ArrayList<>(criteria));
+        this.outputVariables = unmodifiableList(new ArrayList<>(outputVariables));
         this.filter = filter;
         this.leftHashVariable = leftHashVariable;
         this.rightHashVariable = rightHashVariable;
         this.distributionType = distributionType;
-        this.dynamicFilters = ImmutableMap.copyOf(dynamicFilters);
+        this.dynamicFilters = unmodifiableMap(new HashMap<>(dynamicFilters));
```

Comment:
Should it be using LinkedHashMap to preserve order, which was guaranteed by ImmutableMap?